llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1758,7 +1786,15 @@ void AddressSanitizer::instrumentAddress(Instruction *OrigIns,
   size_t Granularity = 1ULL << Mapping.Scale;
   Instruction *CrashTerm = nullptr;
 
-  if (ClAlwaysSlowPath || (TypeStoreSize < 8 * Granularity)) {
+  bool GenSlowPath = (ClAlwaysSlowPath || (TypeStoreSize < 8 * Granularity));
+
+  if (TargetTriple.isAMDGPU()) {
+    if (GenSlowPath) {
```

Comment:
Don't you need 
```
} else {
```
from line 1815?

Reply:
Not really because of the CFG flattening. _If/then_ for report block is created the same either for the condition requiring "slow path" check (it is anded with the loaded shadow memory byte check) or for the condition on the "fast path".

Reply:
Why not llvm-project/llvm/test/Instrumentation/AddressSanitizer/AMDGPU/